src/test/java/de/tum/in/www1/artemis/service/exam/ExamServiceTest.java

```diff
@@ -119,10 +119,9 @@ void validateForStudentExamGeneration_differentCalculationTypesInExerciseGroup_s
 
         exerciseGroup.setExercises(Set.of(includedTextExercise, notIncludedTextExercise));
 
-        BadRequestAlertException thrown = assertThrows(BadRequestAlertException.class, () -> examService.validateForStudentExamGeneration(exam),
-                "Expected to throw bad request alert exception, but it didn't");
-
-        assertThat(thrown).hasMessageContaining("All exercises in an exercise group must have the same meaning for the exam score");
+        assertThatExceptionOfType(BadRequestAlertException.class).as("Expected to throw bad request alert exception, but it didn't")
```

Comment:
I see that you just translated the statement 1:1, but it might be worth considering *in the future*, whether we want to have such statements describing the nature of the exception when failing or not.

Reply:
What exactly do you mean by this?

Reply:
I'm referring to the `.as(...)` section, which provides an error message in case the assertion fails.

We use this, however, quite inconsistently. So I think it makes sense to develop a plan in the future whether to use them at all and if yes, in which cases.

Cause this error message, for example, is pretty useless as it's clear from the assertion itself.

Reply:
I see, for this case I definetly agree, the `as` here is not needed. This can be done in the future after we agree on how to use `as` consistently :)

Reply:
Exactly. Was only meant as a future suggestion :)